src/main/resources/org/biouno/unochoice/stapler/unochoice/UnoChoice.es6

```diff
@@ -144,19 +151,19 @@ var UnoChoice = UnoChoice || ($ => {
      *
      * @param avoidRecursion {boolean} flag to decide whether we want to permit self-reference parameters or not
      */
-    CascadeParameter.prototype.update = function(avoidRecursion) {
+    CascadeParameter.prototype.update = async function(avoidRecursion) {
+        incrementLoadingCounter();
         let parametersString = this.getReferencedParametersAsText(); // gets the array parameters, joined by , (e.g. a,b,c,d)
         console.log(`Values retrieved from Referenced Parameters: ${parametersString}`);
         // Update the CascadeChoiceParameter Map of parameters
-        this.proxy.doUpdate(parametersString);
+        await new Promise((resolve) => this.proxy.doUpdate(parametersString, t => resolve(t)));
         // Now we get the updated choices, after the Groovy script is eval'd using the updated Map of parameters
         // The inner function is called with the response provided by Stapler. Then we update the HTML elements.
         let _self = this; // re-reference this to use within the inner function
         console.log('Calling Java server code to update HTML elements...');
-        this.proxy.getChoicesForUI(t => {
-            let choices = t.responseText;
-            console.log(`Values returned from server: ${choices}`);
-            let data = JSON.parse(choices);
+        await new Promise((resolve) => this.proxy.getChoicesForUI(t => {
+            console.log('Response from server', t);
+            let data = t.responseJSON ? t.responseJSON : JSON.parse(t.responseText);
```

Comment:
`responseJSON` and `responseText` are implementation details of the old Prototype implementation that are only partially supported for backward compatibility. In the case of this plugin, the server is always returning JSON, so the officially supported API can be used:

```suggestion
            let data = t.responseObject();
```

This should work with both old and new versions of core.